python/ray/dag/dag_node_operation.py

```diff
@@ -32,13 +47,50 @@ def __init__(
                 as bind_index because there may be more tasks bound to an actor
                 than tasks that appear in the current compiled DAG.
             operation_type: The type of operation to perform.
+            method_name: The name of the method that this operation originates
+                from. This is only for debugging purposes.
         """
         self.exec_task_idx = exec_task_idx
         self.type = operation_type
+        self.method_name = method_name
+
+    def next_operation(self):
+        """
+        The expected next operation of the same task to perform after this operation.
+
+        If the current operation is READ, this function returns COMPUTE with the same
+        `exec_task_idx` and method name. If the current operation is COMPUTE, this
+        function returns WRITE with the same `exec_task_idx` and method name.
+        Calling this function on a WRITE operation will raise a ValueError.
+        """
+        if self.type == _DAGNodeOperationType.READ:
+            return _DAGNodeOperation(
+                self.exec_task_idx, _DAGNodeOperationType.COMPUTE, self.method_name
+            )
+        elif self.type == _DAGNodeOperationType.COMPUTE:
+            return _DAGNodeOperation(
+                self.exec_task_idx, _DAGNodeOperationType.WRITE, self.method_name
+            )
+        else:
+            raise ValueError(
+                "Cannot only get next operation for READ or COMPUTE type, "
```

Comment:
typo?
```suggestion
                "Can only get next operation for READ or COMPUTE type, "
```

Reply:
This method is not needed any more. Removed.